### PR TITLE
tests/lib/prepare.sh: use ukify instead of llvm-objcopy

### PR DESCRIPTION
**Pull request by valentindavid**

### CODE REVIEW

File: tests/lib/prepare.sh [hunk 1]
```diff
@@ -825,35 +825,29 @@ uc24_build_initramfs_kernel_snap() {
 
     unmkinitramfs initrd.img initrd
 
-    local output_initrd="${PWD}/initrd.img"
-
-    local unpacked_initrd_root="${PWD}/initrd"
-    if os.query is-pc-amd64; then
-        unpacked_initrd_root="${unpacked_initrd_root}/main"
+    if [ -d ./extra-initrd ]; then
```
Comment (ernestl) [inline tests/lib/prepare.sh:828]: Is this currently exercised, seems to not exist for e.g. 24/edge kernel?

**Comment by zyga** [comment]
@valentindavid what'w the backstory on this?

**Comment by valentindavid** [comment]
> @valentindavid what'w the backstory on this?

Originally we used objcopy from binutils, with specific addresses to put the kernel and initrd into the stub. But that did not work great. Then we started to use llvm-objcopy, which could deal with addresses itself. The problem right now is that llvm

But anyway, systemd had the same problems. And they made a nice python script that can build the uki from the stub. And it does exactly what we need. And it is even simpler to use.

**Review by ernestl — APPROVED** [review]
Thanks

**Review by alfonsosanchezbeato — COMMENTED** [review]
`ukify` is only available in 22.04 onwards, is this running only on it? Even if that is the case, probably worth adding a comment about this.

**Comment by valentindavid** [comment]
> `ukify` is only available in 22.04 onwards, is this running only on it? Even if that is the case, probably worth adding a comment about this.

It is only for uc 24 here.

**Pull request merged**